external/CMakeLists.txt

```diff
@@ -29,12 +26,29 @@ get_filename_component(compilername ${CMAKE_MAKE_PROGRAM} NAME)
 
 if(NOT N EQUAL 0)
     if (${compilername} STREQUAL "MSBuild.exe")
-        set(EXTERNAL_BUILD_FLAGS -- /m:${N})
+        set(EXTERNAL_BUILD_FLAGS "-- /m:${N}")
     elseif(NOT (CMAKE_GENERATOR STREQUAL Xcode))
-        set(EXTERNAL_BUILD_FLAGS -- -j${N})
+        set(EXTERNAL_BUILD_FLAGS "-- -j${N}")
     endif()
 endif()
 
+set(COMMON_PLATFORM_FLAGS)
+
+if(ANDROID)
+  set(COMMON_PLATFORM_FLAGS -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE} 
+                            -DANDROID_PLATFORM=${ANDROID_PLATFORM}
+                            -DANDROID_STL=${ANDROID_STL}
+                            -DANDROID_ABI=${ANDROID_ABI})
+elseif(IOS)
+  set(COMMON_PLATFORM_FLAGS -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+                            -DPLATFORM=${PLATFORM}
+                            -DSIMULATOR=${SIMULATOR})
+endif()
+
+if(BUILD_SHARED_LIBS)
+  set(COMMON_PLATFORM_FLAGS ${COMMON_PLATFORM_FLAGS} -DCMAKE_POSITION_INDEPENDENT_CODE=ON)
```

Comment:
Are you sure we need to enable it? Looks like by default, it should be turned on:
https://cmake.org/cmake/help/v3.4/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE :

> This property is True by default for SHARED and MODULE library targets and False otherwise.